### ISSUE TITLE
Regression on GEMM allocation

### ISSUE THREAD

**Issue by mratsim** [numforge/laser#27]
The gemm float32 benchmark is segfaulting with the following stacktrace
```
Traceback (most recent call last)
gemm_bench_float32.nim   gemm_bench_float32
gemm_bench_float32.nim   benchLaserGEMM
gemm.nim                 gemm_strided
gemm_tiling.nim          newTiles
alloc.nim                alloc
alloc.nim                rawAlloc
alloc.nim                getBigChunk
alloc.nim                removeChunkFromMatrix2
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

This happens with or without openmp.

The offending commit is e898f027e6d08d90542714234629da984d4e4639 (#26)

**Comment by mratsim** [comment]
Commenting out extras bench like OpenBLAS and compiling with `nim cpp -r -d:useSysAssert -d:useGcAssert --stacktrace:on -d:release -o:build/gemm_f32_omp benchmarks/gemm/gemm_bench_float32.nim`

we get `[SYSASSERT] dealloc: object header corrupted` (https://github.com/numforge/laser/commit/e898f027e6d08d90542714234629da984d4e4639)
or `[SYSASSERT] result wrong pointer!` (current master https://github.com/numforge/laser/commit/c4ac93da91a1de5c50bf9b5f776ba80b09570759) depending on the commit we are on.

**Comment by mratsim** [comment]
It can be reproduced with full debug:

```
A matrix shape: (M: 1920, N: 1920)
B matrix shape: (M: 1920, N: 1920)
Output shape: (M: 1920, N: 1920)
Required number of operations: 14155.776 millions
Required bytes:                   29.491 MB
Arithmetic intensity:            480.000 FLOP/byte
Theoretical peak single-core:    224.000 GFLOP/s
Theoretical peak multi:         4032.000 GFLOP/s
Make sure to not bench Apple Accelerate or the default Linux BLAS.
15052863
3932223
15052863
3932223
15052863
3932223
15052863
3932223
Traceback (most recent call last)
gemm_bench_float32.nim(346) gemm_bench_float32
gemm_bench_float32.nim(181) benchLaserGEMM
gemm.nim(215)            gemm_strided
gemm_tiling.nim(341)     newTiles
alloc.nim(929)           alloc
alloc.nim(795)           rawAlloc
alloc.nim(596)           getBigChunk
alloc.nim(211)           removeChunkFromMatrix2
SIGSEGV: Illegal storage access. (Attempt to read from nil?
```

The crash is in the second allocShare at the start of the 6th benchmark iteration

**Comment by mratsim** [comment]
Fixed in #28, the newTiles proc wasn't getting mc, nc and kc from partitionMNK in the correct order
